ILSpy/Util/ShellHelper.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace ICSharpCode.ILSpy.Util
+{
+	static class ShellHelper
+	{
+		[DllImport("shell32.dll", CharSet = CharSet.Unicode)]
+		static extern int SHParseDisplayName([MarshalAs(UnmanagedType.LPWStr)] string pszName, IntPtr pbc, out IntPtr ppidl, uint sfgaoIn, out uint psfgaoOut);
+
+		[DllImport("shell32.dll")]
+		static extern int SHOpenFolderAndSelectItems(IntPtr pidlFolder, uint cidl, [MarshalAs(UnmanagedType.LPArray)] IntPtr[] apidl, uint dwFlags);
+
+		[DllImport("shell32.dll")]
+		static extern IntPtr ILFindLastID(IntPtr pidl);
+
+		[DllImport("ole32.dll")]
+		static extern void CoTaskMemFree(IntPtr pv);
+
+		public static void OpenFolder(string folderPath)
+		{
+			try
+			{
+				if (string.IsNullOrEmpty(folderPath))
+					return;
+				if (!Directory.Exists(folderPath))
+					return;
+
+				IntPtr folderPidl = IntPtr.Zero;
+				uint attrs;
+				int hr = SHParseDisplayName(folderPath, IntPtr.Zero, out folderPidl, 0, out attrs);
+				if (hr == 0 && folderPidl != IntPtr.Zero)
+				{
+					SHOpenFolderAndSelectItems(folderPidl, 0, null, 0);
+					CoTaskMemFree(folderPidl);
+				}
+				else
+				{
+					// fallback
+					Process.Start(new ProcessStartInfo { FileName = folderPath, UseShellExecute = true });
+				}
+			}
+			catch
+			{
+				// ignore
```

Comment:
what kind of exceptions are we ignoring here? "catch all" is considered bad pratice.